pandas/core/arrays/arrow/array.py

```diff
@@ -1343,15 +1342,21 @@ def _mode(self: ArrowExtensionArrayT, dropna: bool = True) -> ArrowExtensionArra
             data = self._data
 
         modes = pc.mode(data, pc.count_distinct(data).as_py())
-        values = modes.field(0)
         counts = modes.field(1)
         # counts sorted descending i.e counts[0] = max
+        if not dropna and self._data.null_count > counts[0].as_py():
+            return type(self)(pa.array([None], type=pa_type))
```

Comment:
I just wanted to comment this on the arrow issue that you could do something like the above as a workaround, but so you are already doing that;) 
Personally, I think it's a fine workaround on the pandas side (certainly on the short term). It should also be basically as performant compared to when pyarrow would do this in `mode` itself, since `null_count` is cheap (and cached).